### PR TITLE
[WFLY-11318] Validate requirement for modules previously exported by javax.ejb.api on org.jboss.ws.common

### PR DESCRIPTION
**Pull request by yersan**
Removes from `org.jboss.ws.common` modules that were previously exported by `javax.ejb.api`.

- `javax.xml.rpc.api` is still required by classes in jbossws-common.jar, but the other modules can be removed
- None of the removed modules exports services, so looks fine for any service loader executed in  jbossws-common.jar
- I cannot find an external configuration that could require a class exported by the removed modules.  @asoldano , could you also take a look to this point when you have a chance?

Jira issue: https://issues.jboss.org/browse/WFLY-11318

Additional notes: 
From my analisys, it looks like these dependencies are also unused `org.apache.xerces` and `javax.xml.stream.api`, but I'm not sure yet if they can be removed.

### CODE REVIEW

**Comment by asoldano** [comment]
@yersan thanks for PR. We need to run the jbossws-cxf testsuite against a WildFly instance patched with the changes here and see if all tests are still passing. @jimma / @rsearls can any of perform such verification? (similar changes to be applied on the module.xml for the jbossws-common module in jbossws-cxf sources)

**Comment by rsearls** [comment]
I've tested this change in /home/rsearls/j1/wfly10/wildfly/dist/target/wildfly-15.0.0.Alpha1-SNAPSHOT/ with jboss-cxf master.   All integration tests run successfully. 
Created corresponding jira, https://issues.jboss.org/browse/JBWS-4149.  Awating results of travis CI before submitting PR.

**Comment by asoldano** [comment]
Thanks @rsearls . Actually, I've checked what is preventing the removal of the javax.xml.rpc.api module dependency too in jbossws-common; it's the InvocationHandlerJAXRPC class which I have a feeling was left there for the transition period during which we deprecated and then removed JAX-RPC support... and then it was forgot. So Rebecca, please create a JBWS jira and to get rid of that class as well as the dependency here. Thanks.

**Comment by rsearls** [comment]
I've removed class InvocationHandlerJAXRPC from jbossws-common.
https://issues.jboss.org/browse/JBWS-4150
PR: https://github.com/jbossws/jbossws-common/pull/15.
I removed dependency javax.xml.rpc.api from my local wfly and tested jbossws-cxf.
All tests pass.

@yersan can remove dependency javax.xml.rpc.api from this PR once PR-15 is merged.

**Comment by yersan** [comment]
We have to wait for the next release and component upgrade in wildfly of jbossws before removing `javax.xml.rpc.api`. I assume jbossws/jbossws-common#15 is going to be merged for jbossws-3.2.3.Final.

@rsearls @asoldano Are we going to have a jbossws component upgrade including jbossws/jbossws-common#15 for wildfly 15?

**Comment by kabir** [comment]
@yersan I am adding the hold label until we have the needed upgrade

**Comment by jamezp** [comment]
retest this please

**Comment by jamezp** [comment]
Given it's been some time since this was tested against JBossWS @rsearls can you re-verify this?

**Comment by rsearls** [comment]
jbossws/jbossws-common#15  was merged Nov 20, 2018.
I have retested with wfly-18.0.0.Beta1-SNAPSHOT which is using jbossws-cxf-5.3.0.Final and has the merged code ( jbossws/jbossws-common#15) AND I installed this PR's code change in wfly.
 THe jbossws-cxf testsuite runs successfully.

**Comment by jamezp** [comment]
Awesome. Thanks @rsearls

**Review by jamezp — APPROVED** [review]

**Comment by yersan** [comment]
Thank you both! I did not notice we already had 3.2.3.Final

**Pull request merged**
